kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/IcebergSinkConfig.java

```diff
@@ -0,0 +1,478 @@
+              PropertyUtil.propertiesWithPrefix(originalProps, TABLE_PROP_PREFIX + tableName + ".");
+
+          String routeRegexStr = tableConfig.get(ROUTE_REGEX);
+          Pattern routeRegex = routeRegexStr == null ? null : Pattern.compile(routeRegexStr);
+
+          String idColumnsStr = tableConfig.getOrDefault(ID_COLUMNS, tablesDefaultIdColumns());
+          List<String> idColumns = stringToList(idColumnsStr, ",");
+
+          String partitionByStr =
+              tableConfig.getOrDefault(PARTITION_BY, tablesDefaultPartitionBy());
+          List<String> partitionBy = stringToList(partitionByStr, COMMA_NO_PARENS_REGEX);
+
+          String commitBranch =
+              tableConfig.getOrDefault(COMMIT_BRANCH, tablesDefaultCommitBranch());
+
+          return new TableSinkConfig(routeRegex, idColumns, partitionBy, commitBranch);
+        });
+  }
+
+  @VisibleForTesting
+  static List<String> stringToList(String value, String regex) {
+    if (value == null || value.isEmpty()) {
+      return ImmutableList.of();
+    }
+
+    return Arrays.stream(value.split(regex)).map(String::trim).collect(toList());
+  }
+
+  public String tablesCdcField() {
+    return getString(TABLES_CDC_FIELD_PROP);
+  }
+
+  public String controlTopic() {
+    return getString(CONTROL_TOPIC_PROP);
+  }
+
+  public String controlGroupId() {
+    String result = getString(CONTROL_GROUP_ID_PROP);
+    if (result != null) {
+      return result;
+    }
+    String connectorName = connectorName();
+    Preconditions.checkNotNull(connectorName, "Connector name cannot be null");
+    return DEFAULT_CONTROL_GROUP_PREFIX + connectorName;
+  }
+
+  public String connectGroupId() {
+    String result = getString(CONNECT_GROUP_ID_PROP);
+    if (result != null) {
+      return result;
+    }
+
+    String connectorName = connectorName();
+    Preconditions.checkNotNull(connectorName, "Connector name cannot be null");
+    return "connect-" + connectorName;
+  }
+
+  public int commitIntervalMs() {
+    return getInt(COMMIT_INTERVAL_MS_PROP);
+  }
+
+  public int commitTimeoutMs() {
+    return getInt(COMMIT_TIMEOUT_MS_PROP);
+  }
+
+  public int commitThreads() {
+    return getInt(COMMIT_THREADS_PROP);
+  }
+
+  public String hadoopConfDir() {
+    return getString(HADDOP_CONF_DIR_PROP);
+  }
+
+  public boolean upsertModeEnabled() {
+    return getBoolean(TABLES_UPSERT_MODE_ENABLED_PROP);
+  }
+
+  public boolean autoCreateEnabled() {
+    return getBoolean(TABLES_AUTO_CREATE_ENABLED_PROP);
+  }
+
+  public boolean evolveSchemaEnabled() {
+    return getBoolean(TABLES_EVOLVE_SCHEMA_ENABLED_PROP);
+  }
+
+  public boolean schemaForceOptional() {
+    return getBoolean(TABLES_SCHEMA_FORCE_OPTIONAL_PROP);
+  }
+
+  public boolean schemaCaseInsensitive() {
+    return getBoolean(TABLES_SCHEMA_CASE_INSENSITIVE_PROP);
+  }
+
+  public JsonConverter jsonConverter() {
+    return jsonConverter;
+  }
+
+  private Map<String, String> loadWorkerProps() {
```

Comment:
Can you add a comment explaining what is happening here?

Reply:
I added a comment for this.